### DISCUSSION TITLE
How to Develop Drawer Type Containers

### DISCUSSION THREAD

**Discussion by 2524268884** [mui/toolpad#2773] [Help and support]
Hello! We have currently deployed a set of ToolPads on our own servers, and we have also developed some custom components to meet our internal needs. We currently want to use ToolPad to develop a CRM system to meet our customer management requirements. We have customized a DataGird component to implement data addition, deletion, modification, and query functions. When users click Add, they can fill out forms in a drawer, as shown in the following figure:
![image](https://github.com/mui/mui-toolpad/assets/80509510/9a153b42-8d0a-4dbc-9ef6-ae5a325e42d9)
But we hope to make this drawer component independent, just like the basic container component of ToolPad. We can customize drag and drop to add components to the drawer to meet some of our requirements (such as configuring mandatory attributes for input boxes). At the same time, we can control the display or hiding of this drawer component through other components, which is very necessary for us. Do we have plans to develop similar functions?
Also attached are some references for other platform drawer components:
![image](https://github.com/mui/mui-toolpad/assets/80509510/36501589-7e6b-487b-8570-c6a63dc36c5d)
<img width="1181" alt="image" src="https://github.com/mui/mui-toolpad/assets/80509510/48ee99db-2495-44b6-9971-8d0804a851e3">

**Comment by Janpot** [comment]
Thank you for your feedback, we plan to add this feature. I'm adding a [feature request](https://github.com/mui/mui-toolpad/issues/2775) for a drawer component. Please upvote so we can track interest.

**Reply by 2524268884** [reply]
Thank you for your support! I also hope to consider the issue of component linkage when developing drawer components, as I usually control the display of drawers through button components. Therefore, I may need to be able to control the corresponding properties of this drawer component through JS methods
